### ISSUE TITLE
Pull Request Reviews (Preview API) methods have unusual error format that is not parsed.

### ISSUE THREAD

**Issue by dmitshur** [google/go-github#540]
[Pull Request Reviews (Preview API)](https://developer.github.com/changes/2016-12-14-reviews-api/) methods were tracked in #495 and implemented in #497. They currently seem to use an unusual format for errors, and go-github does not parse that format.

Originally discovered by @haya14busa during review of #497:

> As I posted request and error response example (https://github.com/google/go-github/pull/497#discussion_r92933362), error format for review API is not same as http://developer.github.com/v3/#client-errors
> 
> "errors" is list of string for review API.
> 
> ```json
> {
>   "message": "Validation Failed",
>   "errors": [
>     "Cannot return null for non-nullable field PullRequestReviewComment.position"
>   ],
>   "documentation_url": "https://developer.github.com/v3/pulls/reviews/#get-a-single-reviews-comments"
> }
> ```
> 
> Corresponding go-github struct (`ErrorResponse`)  is here. https://github.com/google/go-github/blob/fbc15caccf54600900355e302a0fafabf7e2d733/github/github.go#L463
> 
> Because `ErrorResponse` doesn't support review API error format, error information is dropped with unmarshalling and it's really difficult to debug error using go-github.
> (I tried to implement create review API in my personal project https://github.com/haya14busa/reviewdog/pull/63 using go-github, but it was really hard to debug error... )
> 
> I'll report this problem to GitHub support, but this review API error fromat could be intentional.
> If so, we have to handle this problem I guess.

See https://github.com/google/go-github/pull/497#issuecomment-267810912 for full context and details.

This issue is to track that problem (which will soon be merged into `master`) and resolve it afterwards. The unusual error format is not documented, and we're waiting on GitHub to potentially fix the error format for those endpoints, or update documentation.

**Comment by dmitris** [comment]
@dmitshur - have you heard anything back from the GitHub support on the issue (since it is marked "wairing for GitHub")?

**Comment by dmitris** [comment]
https://godoc.org/github.com/google/go-github/github#PullRequestsService.ListReviewComments contains a link to this issue:

> TODO: Follow up with GitHub support about an issue with this method's returned error format and remove this comment once it's fixed. Read more about it here - https://github.com/google/go-github/issues/540

**Comment by dmitshur** [comment]
It was @haya14busa that emailed GitHub, not me. See https://github.com/google/go-github/pull/497#issuecomment-270400412.

The latest status in that thread seems to be https://github.com/google/go-github/pull/497#pullrequestreview-19529473.

@haya14busa Can you let us know please if there have been any updates from GitHub since then?

**Comment by haya14busa** [comment]
I pinged 2-3 times but got no clear response from the supports.
Can anybody send a report again? 
Maybe they think it's not high priority issue (or didn't understand the point from my email).

**Comment by Xorima** [comment]
A bit late but with this change does this mean the docs can be updated as well? https://pkg.go.dev/github.com/google/go-github/v39/github#PullRequestsService.CreateReview
